azkaban-webserver/src/main/java/azkaban/webapp/servlet/ExecutorServlet.java

```diff
@@ -341,31 +339,16 @@ private void handleExecutionFlowPage(HttpServletRequest req,
       return;
     }
     
-    Props props = getApplication().getServerProps(); 
-    String execExternalLinkURL = 
-        props.getString("execution.external.link.url", "");
-    String execExternalLinkLabel = 
-        props.getString("execution.external.link.label", "External Analyzer");
-        
-    if (StringUtils.isNotEmpty(execExternalLinkURL)) {
-      StringBuilder builder = new StringBuilder();
-      builder.append(req.getRequestURL());
-      builder.append("?");
-      builder.append(req.getQueryString());
-      String flowExecutionURL = builder.toString();
-      String execExternalLinkURLFormatted = 
-          String.format(execExternalLinkURL, flowExecutionURL);
-      page.add("executionExternalLinkURL", execExternalLinkURLFormatted);
-      page.add("executionExternalLinkLabel", execExternalLinkLabel);
-    }
-
+    Props props = getApplication().getServerProps();
+    ExternalAnalyzerUtils.addExternalAnalyzer(props, page, execId);
```

Comment:
What if you structure the code so that the method will return a URL string to be inserted into the page? Your test will be able to assert the string has the correct content. And your test will not have to deal with page object.